atomium/includes/classes/AttributesContainer.php

```diff
@@ -8,55 +8,72 @@
  * @package Drupal\atomium
  */
 class AttributesContainer implements \ArrayAccess {
+
   /**
    * Stores the attribute data.
    *
-   * @var array
+   * @var \Drupal\atomium\Attributes[]
+   *   Format: $[$channel] = $attributes
    */
   protected $storage = array();
 
   /**
    * AttributesContainer constructor.
    *
-   * @param array $attributes
-   *   An array of attributes.
+   * @param mixed[][] $attributess
+   *   Format:
+   *   - $[$channel][$attribute_name] = $attribute_value :string
+   *   - $[$channel][$attribute_name][] = $attribute_value_part
+   *   - $[$channel][$attribute_name] = FALSE|TRUE  (attributes without value)
    */
-  public function __construct(array $attributes = array()) {
-    foreach ($attributes as $name => $value) {
-      $this->offsetGet($name)->setAttributes($value);
+  public function __construct(array $attributess = array()) {
```

Comment:
'attribustess' is not a very representative name for the variable, I would prefer it if we call it attributes_array, or call single_attributes to the individual attributes variable. In any case, I think there is room for improvement :)

Reply:
This is why it has 2 s :-) It's more than one 'attributes'.